utils/tf.py

```diff
@@ -67,40 +72,52 @@ def __create_config(self, intra_threads: int, inter_threads=1):
         :return: TensorFlow config
         """
         config = tf.compat.v1.ConfigProto()
-        config.allow_soft_placement = True
+        config.log_device_placement = False
         config.intra_op_parallelism_threads = intra_threads
         config.inter_op_parallelism_threads = inter_threads
         return config
 
-    def __initialize_graph(self, path_to_model: str):
+    def __initialize_graph(self, path_to_model: str, use_opsgraph=False):
         """
         A function initializing TF graph from frozen .pb model.
         :param path_to_model: str
         :return: TensorFlow graph
         """
-        graph = tf.compat.v1.Graph()
-        with graph.as_default():
+        if use_opsgraph:
+            graph = ops.Graph()
+            graph_def = graph_pb2.GraphDef()
+
+        else:
+            graph = tf.compat.v1.Graph()
             graph_def = tf.compat.v1.GraphDef()
+
+        with graph.as_default():
             with tf.compat.v1.gfile.GFile(path_to_model, 'rb') as fid:
                 serialized_graph = fid.read()
                 graph_def.ParseFromString(serialized_graph)
-                tf.compat.v1.import_graph_def(graph_def, name="")
+                if use_opsgraph:
+                    tf.import_graph_def(graph_def)
+                else:
+                    tf.compat.v1.import_graph_def(graph_def, name="")
+
         return graph
 
-    def set_input_tensor(self, input_name: str, input_array):
+    def set_input_tensor(self, input_name, input_array):
         """
         A function assigning given numpy input array to the tensor under the provided input name.
         :param input_name: str, name of a input node in a model, eg. "image_tensor:0"
         :param input_array: numpy array with intended input
         """
-        self.__feed_dict[self.__graph.get_tensor_by_name(input_name)] = input_array
+        if isinstance(input_name, str):
+            self.__feed_dict[self.graph.get_tensor_by_name(input_name)] = input_array
+        elif isinstance(input_name, list):
```

Comment:
If many input tensors need to be set then this function should be simply called multiple times from "outside".

Reply:
agreed, I will change it accordingly